src/Module/Contacts.php

```diff
@@ -0,0 +1,1137 @@
+
+			if (!DBA::isResult($contact)) {
+				$contact = DBA::selectFirst('contact', [], ['id' => $contact_id, 'uid' => 0]);
+			}
+
+			// Don't display contacts that are about to be deleted
+			if ($contact['network'] == Protocol::PHANTOM) {
+				$contact = false;
+			}
+		}
+
+		if (DBA::isResult($contact)) {
+			if ($contact['self']) {
+				if (($a->argc == 3) && intval($a->argv[1]) && in_array($a->argv[2], ['posts', 'conversations'])) {
+					goaway('profile/' . $contact['nick']);
+				} else {
+					goaway('profile/' . $contact['nick'] . '?tab=profile');
+				}
+			}
+
+			$a->data['contact'] = $contact;
+
+			if (($a->data['contact']['network'] != "") && ($a->data['contact']['network'] != Protocol::DFRN)) {
+				$networkname = format_network_name($a->data['contact']['network'], $a->data['contact']['url']);
+			} else {
+				$networkname = '';
+			}
+
+			/// @TODO Add nice spaces
+			$vcard_widget = replace_macros(get_markup_template("vcard-widget.tpl"), [
+				'$name' => htmlentities($a->data['contact']['name']),
+				'$photo' => $a->data['contact']['photo'],
+				'$url' => Contact::MagicLink($a->data['contact']['url']),
+				'$addr' => (($a->data['contact']['addr'] != "") ? ($a->data['contact']['addr']) : ""),
+				'$network_name' => $networkname,
+				'$network' => L10n::t('Network:'),
+				'$account_type' => Contact::getAccountType($a->data['contact'])
+			]);
+
+			$findpeople_widget = '';
+			$follow_widget = '';
+			$networks_widget = '';
+		} else {
+			$vcard_widget = '';
+			$networks_widget = Widget::networks('contacts', $nets);
+			if (isset($_GET['add'])) {
+				$follow_widget = Widget::follow($_GET['add']);
+			} else {
+				$follow_widget = Widget::follow();
+			}
+
+			$findpeople_widget = Widget::findPeople();
+		}
+
+		if ($contact['uid'] != 0) {
+			$groups_widget = Group::sidebarWidget('contacts', 'group', 'full', 'everyone', $contact_id);
+		} else {
+			$groups_widget = null;
+		}
+
+		$a->page['aside'] .= replace_macros(get_markup_template("contacts-widget-sidebar.tpl"), [
+			'$vcard_widget' => $vcard_widget,
+			'$findpeople_widget' => $findpeople_widget,
+			'$follow_widget' => $follow_widget,
+			'$groups_widget' => $groups_widget,
+			'$networks_widget' => $networks_widget
+		]);
+
+		$base = System::baseUrl();
+		$tpl = get_markup_template("contacts-head.tpl");
+		$a->page['htmlhead'] .= replace_macros($tpl, [
+			'$baseurl' => System::baseUrl(true),
+			'$base' => $base
+		]);
+	}
+
+	private static function contacts_batch_actions(App $a)
+	{
+		if (empty($_POST['contact_batch']) || !is_array($_POST['contact_batch'])) {
+			return;
+		}
+
+		$contacts_id = $_POST['contact_batch'];
+
+		$orig_records = q("SELECT * FROM `contact` WHERE `id` IN (%s) AND `uid` = %d AND `self` = 0",
+			implode(",", $contacts_id),
+			intval(local_user())
+		);
+
+		$count_actions = 0;
+		foreach ($orig_records as $orig_record) {
+			$contact_id = $orig_record['id'];
+			if (x($_POST, 'contacts_batch_update')) {
```

Comment:
Ohhh it has to be `defaults($_POST, 'contacts_batch_update', '')` here. Otherwise every if case is true and the contact is updated, archived, block, ignored and deleted.